script-reference/attributes.rst

```diff
@@ -566,17 +566,44 @@ warnings.
 &is_used
 --------
 
-Zeek has static analysis capabilities
-for detecting locations in a script where local variables are assigned
-values that are not subsequently used (i.e. "dead code").
-For cases where it's desirable
-to suppress the warning, the ``&is_used`` attribute may be applied, for
-example:
+Zeek has static analysis capabilities for detecting locations in a script where
+local variables are assigned values that are not subsequently used (i.e. "dead
+code").
+
+It can also warn about unused functions, hooks, and event handlers.  The intent
+behind these checks is to catch instances where the script writer has introduced
+typos in names, or has forgotten to remove code that's no longer needed.  For
+functions and hooks, "unused" means the function/hook is not exported or in the
+global scope (nor deprecated), and no "live" (i.e., not "unused")
+function/hook/event handler calls it.  For event handlers, "unused" means that
+the event engine does not generate the event, nor do any "live"
+function/hook/event handler generates (and the event handler is not deprecated).
```

Comment:
```suggestion
function/hook/event handler generate it (and the event handler is not deprecated).
```

Maybe move the deprecated special-case mentions out of parenthesis and into a separate sentence in the beginning or end?

"Functions/hooks/event handlers marked deprecated (:zeek:see:`&deprecated`) are never reported unused."

Reply:
Great suggestion, yeah — that paragraph is hairy. :-) Forthcoming in a force-push.